### PR TITLE
19937 added alternateNames to business json

### PR DESCRIPTION
**Pull request by PaulGarewal**
*Issue #:* /bcgov/entity#19937

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the business-schemas license (Apache 2.0).

### CODE REVIEW

File: src/registry_schemas/example_data/schema_data.py [hunk 1]
```diff
@@ -50,7 +50,7 @@
     'lastLedgerTimestamp': '2019-04-15T20:05:49.068272+00:00',
     'lastPreBobFilingTimestamp': '2019-04-15T20:05:49.068272+00:00',
     'legalName': 'legal name - CP1234567',
-    'businessName': 'legal name - CP1234567',
+    'alternateNames': 'legal name - CP1234567',
```
Comment (PaulGarewal) [inline src/registry_schemas/example_data/schema_data.py:53]: Not sure if I should carry on editing this file, if yes I will continue, if no I will revert these changes

Comment (argush3) [reply src/registry_schemas/example_data/schema_data.py:53]: You can replace both instances of `businessName` with `"alternateNames": []`.

This is what the legal api currently returns when there are no alternate name entries.

File: src/registry_schemas/schemas/business.json [hunk 2]
```diff
@@ -79,6 +72,38 @@
...
+            "type": "string",
+            "format": "date-time",
+            "title": "The timestamp of the registered date.",
+            "default": "",
+            "examples": ["1970-01-01T00:00:00+00:00"]
+          },
+          "nameStartDate": {
+            "type": "string",
+            "format": "date",
+            "title": "The start date of the alternate name Schema",
```
Comment (argush3) [inline src/registry_schemas/schemas/business.json:94]: "Schema" can be removed

**Review by vysakh-menon-aot — APPROVED** [review]

File: src/registry_schemas/schemas/business.json [hunk 3]
```diff
@@ -79,6 +72,38 @@
...
+          },
+          "operatingName": {
+            "type": "string",
+            "title": "The Operating Name of the business.",
+            "default": "",
+            "examples": ["operating_name"],
+            "pattern": "^(.*)$"
+          }
+        }
+      },
```
Comment (argush3) [inline src/registry_schemas/schemas/business.json:106]: Had a discussion with Thor about making some tweaks to `alternateName` and to also update it so we can support what we have under the `aliases` table currently as well.

Here's an example of what the new alternateNames will look like:

```json
{
  "business": {
    "alternateNames": [
      {
        "name": "SOME NAME TRANSLATION 1",
        "nameStartDate": "2022-05-01",
        "nameType": "TRANSLATION"
      },
      {
        "name": "SOME NAME TRANSLATION 2",
        "nameStartDate": "2022-05-03",
        "nameType": "TRANSLATION"
      },
      {
        "entityType": "GP",
        "identifier": "FM1000019",
        "name": "FOX FOOD CHAIN",
        "nameRegisteredDate": "2021-04-28T00:00:00+00:00",
        "nameStartDate": "2022-04-29",
        "nameType": "DBA"
      },
      {
        "entityType": "SP",
        "identifier": "FM1111111",
        "name": "SWIFT FOX SNACKS",
        "nameRegisteredDate": "2023-05-10T00:00:00+00:00",
        "nameStartDate": "2023-05-10"
      }
    ],
    "identifier": "FM1000019",
    "legalName": "FOX2 DO2, FOX3 DO3",
    "legalType": "GP"
  }
}
```

So we'll need to have the following updates to the `alternateName` definition here:
* Add `nameType` enum which can have values of `TRANSLATION` and `DBA`
* Rename `operatingName` to `name`
* Add `nameEndDate`  Note: end dated names will not be returned for a business by default but we need this for scenarios where we will return historical info

**Review by argush3 — APPROVED** [review]

**Comment by argush3** [comment]
looks, good.  thanks for the additional updates.

**Review by thorwolpert — APPROVED** [review]
Not a big fan of stuttering, alternateName. nameRegisteredDate, but ok to go.

**Pull request merged**